### DISCUSSION TITLE
Uncompiled apk files have "wrong" data

### DISCUSSION THREAD

**Discussion by parasiteoflife** [iBotPeaches/Apktool#2868] [General]
I'm uncompiling a tiktok apk downloaded from here: https://www.apkmirror.com/apk/tiktok-pte-ltd/tik-tok/tik-tok-25-7-5-release/tiktok-25-7-5-android-apk-download

But when I'm recompiling the apk I get a bunch of attribute errors and I have to edit the xml files to remove things like duplicated attr in `<selector xmlns="" xmlns="">`, are those errors made by apktool? I mean isn't supposed that what we get are the original untouched files? If so then why do they have been build successfully with those duplicated attrs but apktool fails? Does apktool needs an update or something?

I also got a bunch of `AndroidManifest.xml:1111: error: '@1845624832' is incompatible with attribute resource (attr) reference.` and I had to go to the lines and delete the attributes, I haven't tested the app, I only saw apktool finally compiling it.
Thanks!

**Comment by iBotPeaches** [comment]
>  I mean isn't supposed that what we get are the original untouched files?

If only that was as easy it sounds.

> If so then why do they have been build successfully with those duplicated attrs but apktool fails?

Too many reasons for this - so many techniques exist now to malform an application that works for Android, but seriously affects the ability to return the application to source.

> Does apktool needs an update or something?

With 87 issues and 3 open pull requests - I think it always needs something.

>  I had to go to the lines and delete the attributes, I haven't tested the app, I only saw apktool finally compiling it.

That is good - Apktool can do everything automatically.

**Reply by parasiteoflife** [reply]
just tested the apk and besides removing the duplicated attributes I only changed the app icon and name, the apk is not recognized, when I try to open it so it shows me the name and version nothing happens, using external installers also nothing happens, maybe it needed the duplicate attributes to work? but still I don't understand why the package installer fails to show even the name and version of the apk file

**Comment by RenateUSB** [comment]
There can be bad stuff in the actual binary XML file. It certainly is possible to duplicate attributes.
The fault is probably in the actual source but the Android build tools have propagated the error to the binary XML.
I was `apktool d`-ing an app and it made bad XML. There was a spurious "> (as two raw chars) after a closing element.
By looking at the binary XML I could see that the problem was right there and even the line number (30).

```
000000  XML
000008  String pool[14]
000194  XML resource map
0001cc    17  com.android.systemui.stackdivider.DividerView
0001f0        │  android:layout_width=-1
000204        │  android:layout_height=-1
000218    22  ├──View
00023c        │     android:id=@7F0A0186
000250        │     android:background=@7F060065
000264        │     style=@7F1200BB
000290    27  ├──com.android.systemui.stackdivider.MinimizedDockShadow
0002b4        │     android:id=@7F0A02B8
0002c8        │     android:alpha=0.0
0002dc        │     style=@7F1200BD
000308    30  ├──"&quot;&gt; "
000324    32  └──com.android.systemui.stackdivider.DividerHandleView
000348              android:id=@7F0A0187
00035c              android:background=@00000000
000370              android:contentDescription=@7F110065
000384              style=@7F1200BC
0003e0
```
I'm too lazy to download 150 MB of TikTok, but if you post the double ns binary XML I'll look at it.
